src/SortableTrait.php

```diff
@@ -219,10 +221,41 @@ public function moveToEnd()
         $this->$orderColumnName = $maxOrder;
         $this->save();
 
-        static::where($this->getKeyName(), '!=', $this->id)
+        static::applySortableGroup($this->newQuery(), $this)->where($this->getKeyName(), '!=', $this->id)
             ->where($orderColumnName, '>', $oldOrder)
             ->decrement($orderColumnName);
 
         return $this;
     }
+
+    /**
+     * @param QueryBuilder        $query
+     * @param Model|SortableTrait $model
+     *
+     * @return QueryBuilder
+     */
+    protected static function applySortableGroup($query, $model)
+    {
+        $sortableGroupField = static::getSortableGroupField();
+
+        if (is_array($sortableGroupField)) {
+            foreach ($sortableGroupField as $field) {
+                $query = $query->where($field, $model->$field);
+            }
+        } elseif ($sortableGroupField !== null) {
+            $query = $query->where($sortableGroupField, $model->$sortableGroupField);
+        }
+
+        return $query;
+    }
+
+    /**
+     * @return string|null
+     */
+    public static function getSortableGroupField()
+    {
+        $sortableGroupField = isset($this->sortable['sort_by_group_column']) ? $this->sortable['sort_by_group_column'] : null;
```

Comment:
This should be `$model->sortable['sort_by_group_column']` not `$this->sortable['sort_by_group_column']`.

Also, since this package requires `PHP >= 7.0` this method can be changed to:
```php
public static function getSortableGroupField($model)
{
    return $model->sortable['sort_by_group_column'] ?? null;
}
```

or just inlined in the`applySortableGroup()` method.